File: app/src/main/java/com/zhuwenhao/demo/fragment/DeviceInfoFragment.java
package com.zhuwenhao.demo.fragment;

import android.content.Context;
import android.net.wifi.WifiInfo;
import android.net.wifi.WifiManager;
import android.os.Build;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.util.DisplayMetrics;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.zhuwenhao.demo.R;
import com.zhuwenhao.demo.utils.NetworkUtils;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

public class DeviceInfoFragment extends Fragment {

    @BindView(R.id.text_model)
    TextView textModel;

    @BindView(R.id.text_manufacturer)
    TextView textManufacturer;

    @BindView(R.id.text_board)
    TextView textBoard;

    @BindView(R.id.text_device)
    TextView textDevice;

    @BindView(R.id.text_product)
    TextView textProduct;

    @BindView(R.id.text_android_version)
    TextView textAndroidVersion;

    @BindView(R.id.text_code_name)
    TextView textCodeName;

    @BindView(R.id.text_api_level)
    TextView textApiLevel;

    @BindView(R.id.text_resolution)
    TextView textResolution;

    @BindView(R.id.text_available_resolution)
    TextView textAvailableResolution;

    @BindView(R.id.text_density)
    TextView textDensity;

    @BindView(R.id.text_exact_density)
    TextView textExactDensity;

    @BindView(R.id.text_network_type)
    TextView textNetworkType;

    @BindView(R.id.text_ssid)
    TextView textSSID;

    @BindView(R.id.text_ip_address)
    TextView textIpAddress;

    @BindView(R.id.text_mac_address)
    TextView textMacAddress;

    Unbinder unbinder;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_device_info, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initView();
    }

    private void initView() {
        textModel.setText(Build.MODEL);
        textManufacturer.setText(Build.MANUFACTURER);
        textBoard.setText(Build.BOARD);
        textDevice.setText(Build.DEVICE);
        textProduct.setText(Build.PRODUCT);

        textAndroidVersion.setText(Build.VERSION.RELEASE);
        textCodeName.setText(getCodeName());
        textApiLevel.setText(String.valueOf(Build.VERSION.SDK_INT));

        DisplayMetrics dm = new DisplayMetrics();
        getActivity().getWindowManager().getDefaultDisplay().getRealMetrics(dm);
        textResolution.setText(dm.widthPixels + " x " + dm.heightPixels + " px");
        getActivity().getWindowManager().getDefaultDisplay().getMetrics(dm);
        textAvailableResolution.setText(dm.widthPixels + " x " + dm.heightPixels + " px");
        textDensity.setText(String.valueOf(dm.densityDpi + " dp"));
        textExactDensity.setText(dm.xdpi + " x " + dm.ydpi + " dp");

        textNetworkType.setText(NetworkUtils.formatNetworkType(getContext()));
        switch (NetworkUtils.getNetworkType(getContext())) {
            case NetworkUtils.TYPE_NONE:
            case NetworkUtils.TYPE_OTHER:
                getActivity().findViewById(R.id.row_ssid).setVisibility(View.GONE);
                getActivity().findViewById(R.id.row_ip_address).setVisibility(View.GONE);
                getActivity().findViewById(R.id.row_mac_address).setVisibility(View.GONE);
                break;
            case NetworkUtils.TYPE_WIFI:
                WifiManager wifiManager = (WifiManager) getActivity().getSystemService(Context.WIFI_SERVICE);
                WifiInfo wifiInfo = wifiManager.getConnectionInfo();
                textSSID.setText(wifiInfo.getSSID().replace("\"", ""));
                textIpAddress.setText(NetworkUtils.formatIpAddress(wifiInfo.getIpAddress()));
                textMacAddress.setText(NetworkUtils.getMacAddress());
                break;
            case NetworkUtils.TYPE_MOBILE:
            case NetworkUtils.TYPE_MOBILE_2G:
            case NetworkUtils.TYPE_MOBILE_3G:
            case NetworkUtils.TYPE_MOBILE_4G:
                getActivity().findViewById(R.id.row_ssid).setVisibility(View.GONE);
                // TODO: 2017/9/22 get IP and MAC
                break;
        }
    }

    private String getCodeName() {
        switch (Build.VERSION.SDK_INT) {
            case 21:
            case 22:
                return "Lollipop";
            case 23:
                return "Marshmallow";
            case 24:
            case 25:
                return "Nougat";
            case 26:
                return "Oreo";
            default:
                return "";
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }
}

File: app/src/main/java/com/zhuwenhao/demo/fragment/DeviceInfoFragment.kt
package com.zhuwenhao.demo.fragment

import android.content.Context
import android.net.wifi.WifiManager
import android.os.Build
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.DisplayMetrics
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.zhuwenhao.demo.R
import com.zhuwenhao.demo.utils.NetworkUtils
import kotlinx.android.synthetic.main.fragment_device_info.*
import java.text.DecimalFormat

class DeviceInfoFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.fragment_device_info, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
    }

    private fun initView() {
        textModel.text = Build.MODEL
        textManufacturer.text = Build.MANUFACTURER
        textBoard.text = Build.BOARD
        textDevice.text = Build.DEVICE
        textProduct.text = Build.PRODUCT

        textAndroidVersion.text = Build.VERSION.RELEASE
        textCodeName.text = getCodeName()
        textApiLevel.text = Build.VERSION.SDK_INT.toString()

        val dm = DisplayMetrics()
        activity.windowManager.defaultDisplay.getRealMetrics(dm)
        textResolution.text = getString(R.string.resolution_value, dm.widthPixels, dm.heightPixels)
        activity.windowManager.defaultDisplay.getMetrics(dm)
        textAvailableResolution.text = getString(R.string.resolution_value, dm.widthPixels, dm.heightPixels)
        textDensity.text = getString(R.string.density_value, dm.densityDpi)
        textExactDensity.text = getString(R.string.exact_density_value, DecimalFormat("0.000").format(dm.xdpi), DecimalFormat("0.000").format(dm.ydpi))

        textNetworkType.text = NetworkUtils.formatNetworkType(context)
        when (NetworkUtils.getNetworkType(context)) {
            NetworkUtils.TYPE_NONE,
            NetworkUtils.TYPE_OTHER -> {
                rowSSID.visibility = View.GONE
                rowIpAddress.visibility = View.GONE
                rowMacAddress.visibility = View.GONE
            }
            NetworkUtils.TYPE_WIFI -> {
                val wifiInfo = (activity.getSystemService(Context.WIFI_SERVICE) as WifiManager).connectionInfo
                textSSID.text = wifiInfo.ssid.replace("\"", "")
                textIpAddress.text = NetworkUtils.formatIpAddress(wifiInfo.ipAddress) //NetworkUtils.getIpAddress() 效果一样
                textMacAddress.text = NetworkUtils.getMacAddress()
            }
            NetworkUtils.TYPE_MOBILE,
            NetworkUtils.TYPE_MOBILE_2G,
            NetworkUtils.TYPE_MOBILE_3G,
            NetworkUtils.TYPE_MOBILE_4G -> {
                rowSSID.visibility = View.GONE
                textIpAddress.text = NetworkUtils.getIpAddress()
                textMacAddress.text = NetworkUtils.getMacAddress()
            }
        }
    }

    private fun getCodeName(): String {
        return when (Build.VERSION.SDK_INT) {
            21, 22 -> "Lollipop"
            23 -> "Marshmallow"
            24, 25 -> "Nougat"
            26 -> "Oreo"
            else -> ""
        }
    }
}